src/lib.rs

```diff
@@ -149,7 +175,18 @@ pub fn include_shader(input: TokenStream) -> TokenStream {
         [TokenTree::Literal(lit)] => unwrap_string_literal(lit),
         _ => panic!("Takes 1 argument and the argument must be a string literal"),
     };
-    let root_path = resolve_path(&arg);
+
+    let root_path;
+    
+    #[cfg(feature = "relative-path")] {
+        let mut file_path = proc_macro::Span::call_site().source_file().path();
+        file_path.pop();
+        root_path = resolve_path(&arg, Some(&file_path));
+    }
+    #[cfg(not(feature = "relative-path"))] {
+        root_path = resolve_path(&arg, None);
+    }
```

Comment:
In order to reduce code complexity and feature flag checks, could we please refacto this part?

```suggestion
    #[allow(unused_assignments)]
    let mut call_parent_dir_path = None;
    
    #[cfg(feature = "relative-path")] {
        let mut path = proc_macro::Span::call_site().source_file().path();
        path.pop();
        call_parent_dir_path = Some(path);
    }

    let root_path = resolve_path(&arg, call_parent_dir_path);
```